mcswupd/main.go

```diff
@@ -0,0 +1,298 @@
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/clearlinux/mixer-tools/builder"
+	"github.com/clearlinux/mixer-tools/helpers"
+	"github.com/clearlinux/mixer-tools/swupd"
+	"github.com/pkg/errors"
+)
+
+var mixWS = "/usr/share/mix"
+
+const builderConf = `[Mixer]
+LOCAL_BUNDLE_DIR = /usr/share/mix/local-bundles
+
+[Builder]
+SERVER_STATE_DIR = /usr/share/mix/update
+BUNDLE_DIR = /usr/share/mix/local-bundles
+YUM_CONF = /usr/share/mix/.yum-mix.conf
+CERT = /usr/share/mix/Swupd_Root.pem
+VERSIONS_PATH =/usr/share/mix
+LOCAL_RPM_DIR = /usr/share/mix/local-rpms
+LOCAL_REPO_DIR = /usr/share/mix/local
+
+[swupd]
+BUNDLE=os-core
+CONTENTURL=file:///usr/share/mix/update/www
+VERSIONURL=file:///usr/share/mix/update/www
+FORMAT=1
+`
+
+func getCurrentVersion() (int, error) {
+	c, err := ioutil.ReadFile("/usr/lib/os-release")
+	if err != nil {
+		return -1, err
+	}
+
+	re := regexp.MustCompile(`(?m)^VERSION_ID=(\d+)\n`)
```

Comment:
I would use `$` rather than `\n` in the expression, just in case the file didn't have a trailing newline.

A text file on unix is either empty or has a newline as the final character, but nothing stops you having a non text file.